akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -169,10 +187,17 @@ class Decoder(
             recipient,
             localAddress, // FIXME: Is this needed anymore? What should we do here?
             deserializedMessage,
-            senderOption,
+            sender,
             headerBuilder.uid)
 
-          push(out, decoded)
+          if (recipient.isEmpty && !headerBuilder.isNoRecipient) {
+            // the remote deployed actor might not be created yet when resolving the
+            // recipient for the first message that is sent to it, best effort retry
```

Comment:
`resolveRecipient` returns `OptionVal.None` when it the recipient is remote deployed and can't be resolved, otherwise it will always return `Some`.

I have also prepared for the case when we have no recipient at all, which we will need for control messages, and where we currently use the `dummy`